lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+        }
+        return this._config;
+    }
+
+    /**
+     * Build the this._config.rules array
+     * @return {object} - contains error if any rule returned an error
+     * or parsing failed
+     */
+    _buildRulesArray() {
+        const rules = {};
+        this._config.rules = [];
+        if (!this._parsedXML || this._parsedXML === '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml is undefined or empty');
+            return rules;
+        }
+        if (!this._parsedXML.LifecycleConfiguration &&
+            this._parsedXML.LifecycleConfiguration !== '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include LifecycleConfiguration');
+            return rules;
+        }
+        const lifecycleConf = this._parsedXML.LifecycleConfiguration;
+        const rulesArray = lifecycleConf.Rule;
+        if (!rulesArray || rulesArray.length === 0) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include at least one rule');
+            return rules;
+        }
+        if (rulesArray.length > 1000) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml includes over max limit of 1000 rules');
+            return rules;
+        }
+        for (let i = 0; i < rulesArray.length; i++) {
+            const rule = this._parseRule(rulesArray[i]);
+            if (rule.error) {
+                rules.error = rule.error;
+            } else {
+                this._config.rules.push(rule);
+            }
+        }
+        return rules;
+    }
+
+    /**
+     * Check that each xml rule is valid
+     * @param {object} rule - a rule object from Rule array from this._parsedXml
+     * @return {object} - contains error if any component returned an error
+     * or parsing failed, else contains parsed rule object
+     *
+     * Format of ruleObj:
+     * ruleObj = {
+     *      ruleID: <value>,
+     *      ruleStatus: <value>,
+     *      filter: {
+     *          rulePrefix: <value>,
+     *          tags: [
+     *              {
+     *                  key: <value>,
+     *                  val: <value>,
+     *              }
+     *          ]
+     *      }
+     *      actions: [
+     *          {
+     *              actionName: <value>,
+     *              day: <value>,
+     *              date: <value>,
+     *              deleteMarker: <value>
+     *          },
+     *      ]
+     * }
+     */
+    _parseRule(rule) {
+        const ruleObj = {};
+        if (rule.Transition || rule.NoncurrentVersionTransition) {
+            ruleObj.error = errors.NotImplemented.customizeDescription(
+                'Transition lifecycle action not yet implemented');
+            return ruleObj;
+        }
+        if ((!rule.Filter && rule.Filter !== '') &&
+        (!rule.Prefix && rule.Prefix !== '')) {
```

Comment:
maybe `(!rule.Filter || rule.Filter === '')` and also for `Prefix`. If `rule.Filter` and `rule.Prefix` are both empty strings, this will pass

Reply:
I know this logic is confusing >.< but it's meant to be that way because the conditional is supposed to pass if either Filter or Prefix is an empty string. I'll add a comment to hopefully clarify a bit